src/agent/util/utils.ts

```diff
@@ -34,6 +37,52 @@ export interface Listener {
 }
 // Exposed for unit testing.
 export function findScripts(
+    scriptPath: string, config: ResolvedDebugAgentConfig, fileStats: ScanStats,
+    logger: consoleLogLevel.Logger): string[] {
+  // (path: string, knownFiles: string[], resolved: string[]) => string[]
+  const resolved = _findScripts(scriptPath, config, fileStats);
+  if (config.pathResolver) {
+    if (!_.isFunction(config.pathResolver)) {
+      logger.warn(
+          `The 'pathResolver' config must be a function.  Continuing ` +
+          `with the agent's default behavior.`);
+      return resolved;
+    }
+
+    const knownFiles = Object.keys(fileStats);
+    const calculatedPaths =
+        config.pathResolver(scriptPath, knownFiles, resolved);
+    if (calculatedPaths === undefined) {
+      return resolved;
+    }
+
+    if (!calculatedPaths || !Array.isArray(calculatedPaths)) {
+      logger.warn(
+          `The 'pathResolver' config function returned a value ` +
+          `other than 'undefined' or an array of strings. Continuing with ` +
+          `the agent's default behavior.`);
+      return resolved;
+    }
+
+    for (const path of calculatedPaths) {
+      if (knownFiles.indexOf(path) === -1) {
+        // tslint:disable-next-line:no-unused-expression
+        logger.warn(
+            `The 'pathResolver' config function returned a path '${path}' ` +
+            `that is not in the list of paths known to the debug agent ` +
+            JSON.stringify(knownFiles, null, 2)) +
```

Comment:
Move last `)` down two lines.

Reply:
`gts` placed that paren there.

Reply:
I do not think that is the reason. The last two strings appear to be erroneously appended to the return result of `logger.warn(arg)` (rather than `arg` itself).

Reply:
Fixed